### PR TITLE
Make dividers one device pixel thick as defined in Material design spec

### LINKED ISSUE #10490: Divider widget's line width does not match Material design guidelines
## Steps to Reproduce

Create a Divider widget and observe how it's rendered. Here is a screenshot from iOS simulator running iPhone 7 in front of Material design documentation page on Dividers:

image

Setting divider height to < 1.0 seems to be restricted by assertion.

The same behavior can be observed in `ListTile.divideTiles` which uses `DecoratedBox` with border decoration of 1.0 width.

I did some experimentation and was able to get _better_ rendering by implementing custom `divideTiles` method and setting border width to 0.5. Result:

image

## Logs

N/A

## Flutter Doctor

N/A

### PR DESCRIPTION
**Pull request by pulyaevskiy**
Fixes #10490 .

@Hixie 

I've updated `Divider`, `DrawerHeader` and `ListTile.divideTiles` and it looks better now on a device. Also updated Divider's build method.

I'm not exactly sure what kind of test cases I can add here, would appreciate any hints!
Added some changes to the docs, let me know if I can make it better.

### CODE REVIEW

**Comment by Hixie** [comment]
A good way to test this would be to use the "mock_canvas" test harness to make sure the width of the line is 0.0. Have a look around the test widgets or material directories for tests that import "mock_canvas" to see what I mean.

**Comment by Hixie** [comment]
cc @HansMuller

**Comment by pulyaevskiy** [comment]
Great, thanks @Hixie ! It's what I was looking for.

**Comment by pulyaevskiy** [comment]
@Hixie @HansMuller 

I pushed updated tests and a small change to the docs.

File: packages/flutter/lib/src/material/divider.dart [hunk 1]
```diff
@@ -26,19 +26,22 @@ import 'theme.dart';
...
+  }) : assert(height >= 0.0),
        super(key: key);
 
   /// The divider's vertical extent.
   ///
-  /// The divider itself is always drawn as one logical pixel thick horizontal
+  /// The divider itself is always drawn as one device pixel thick horizontal
   /// line that is centered within the height specified by this value.
+  ///
+  /// Divider of 0.0 height is drawn as one device pixel line without any
```
Comment (Hixie) [inline packages/flutter/lib/src/material/divider.dart]: "A divider with a height of 0.0 will be drawn as a line with a height of exactly one device pixel, without any padding around it."

**Comment by Hixie** [comment]
LGTM
...with one minor change to the docs.

Thanks!

File: packages/flutter/lib/src/material/divider.dart (hunk 1 above)
Comment (pulyaevskiy) [inline packages/flutter/lib/src/material/divider.dart]: Very minor but just curious: when I was working on a PR for the DartSDK they had a convention to not use future tense in documentation (in particular this [comment](https://github.com/dart-lang/sdk/pull/26012#r75671896)). Is it applicable to Flutter?

Would "A divider with a height of 0.0 is always drawn as a line with a height of exactly one device pixel, without any padding around it." work?

Comment (Hixie) [inline packages/flutter/lib/src/material/divider.dart]: That works.

The rules we have for Flutter docs are here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#documentation-comments-dartdocs

No mention of tense yet! :-)

Comment (pulyaevskiy) [inline packages/flutter/lib/src/material/divider.dart]: And fixed.

**Comment by pulyaevskiy** [comment]
Just checking if there is anything left on my side? I see flutter-build is red but I'm not sure if it's related to this changes...

**Comment by Hixie** [comment]
Nope, everything's great! We're having some issues with our build pipeline that will probably be resolved on Monday, then I'll check it in. Please don't hesitate to bug me if you don't see it get checked in in the next few days!

**Comment by Hixie** [comment]
I'm watching the build, will land when I'm confident the build is reasonably stable.

**Comment by Hixie** [comment]
landed. thanks @pulyaevskiy!

**Comment by pulyaevskiy** [comment]
Nice! Thanks for your help @hixie!

**Pull request merged**